contrib/cirrus/lib.sh

```diff
@@ -3,293 +3,188 @@
 # Library of common, shared utility functions.  This file is intended
 # to be sourced by other scripts, not called directly.
 
+# BEGIN Global export of all variables
+set -a
+
 # Global details persist here
-source /etc/environment  # not always loaded under all circumstances
+source /etc/environment || touch /etc/environment  # needed on Mac OSX
+
+# Under some contexts these values are not set, make sure they are.
+USER="$(whoami)"
+HOME="$(getent passwd $USER | cut -d : -f 6)"
+[[ -n "$UID" ]] || \
+    UID=$(getent passwd $USER | cut -d : -f 3)
+GID=$(getent passwd $USER | cut -d : -f 4)
 
-# Automation environment doesn't automatically load for Ubuntu 18
+# Automation environment doesn't automatically load for non-interactive shells
 if [[ -r '/usr/share/automation/environment' ]]; then
     source '/usr/share/automation/environment'
+    for libname in defaults anchors console_output utils; do
+        # Pre-installed into environment
+        # shellcheck disable=SC1090,SC2154
+        source $AUTOMATION_LIB_PATH/${libname}.sh
```

Comment:
I don't see `$AUTOMATION_LIB_PATH` defined anywhere! What am I missing?

Reply:
and ***I*** see an opportunity for a more helpful comment :grinning: